types/ws/index.d.ts

```diff
@@ -27,6 +27,17 @@ import { SecureContextOptions } from "tls";
 import { URL } from "url";
 import { ZlibOptions } from "zlib";
 
+type FirstArgument<F extends typeof Buffer.from> = F extends (data: infer A) => any ? A : never;
+
+type BufferLike =
```

Comment:
This list misses
- number
- Array
- DataView (unless this is the same as ArrayBufferView)
- TypedArray (which sounds like Uint8Array and friends?)
- many things with { [Symbol.toPrimitive]() } or with { valueOf() }, since often these aren't declared in TS types.

Reply:
I'm not very sure about this list…

You can find `{[Symbol.toPrimitive]:…}` is tested and it pass.

And I write a union `| { valueOf(): string }`

Reply:
it something like `{[Symbol.toPrimitive]: () => TypedArray}`

Reply:
I got it, these three comments are one same problme

Reply:
and `TypedArray` (including "friends") is `ArrayBufferView` in ts. `DataView` is just `DataView`

Reply:
Thanks for the explanations of DataView and TypedArray.

What I mean for Symbol.toPrimitive and valueOf is that many values may have those methods at runtime, but the Typescript type won't include them. So if there is existing code that doesn't have those methods in their types, it relies on assignability to `any`.

Reply:
I don't get what your are suggesting…

Reply:
People could have an object that supports `valueOf()` at runtime, but isn't present on the type. I *thought* this was common in Typescript because `interface Object` doesn't include `valueOf`. But it actually does.

So...my objection was based on my misunderstanding. Now, I actually think that valueOf will be present in the type system, so it makes sense to require `{ valueOf(): string }`.